appmap/django.py

```diff
@@ -224,12 +224,20 @@ def __call__(self, request):
             self.on_exception(rec, start, call_event_id, sys.exc_info())
             raise
 
+        # response.headers attribute is missing here while running some
+        # django-oscar project tests. There is a response._header attribute
+        # instead.
+        #   django-oscar commit hash: 9574ce2a56f3d99836c2e20785c116da12af6c42
+        #   example test: tests/functional/dashboard/test_dashboard.py
+        #                 TestDashboardIndexForAnonUser::test_is_not_available
+        headers = getattr(response, 'headers', getattr(response, '_headers', None))
```

Comment:
Why does trying to use `headers` instead of `_headers` cause the event-ordering problem?

Reply:
While running a `django-oscar` test, here "AttributeError: 'HttpResponseRedirect' object has no attribute 'headers'" is raised in our code. `pytest` framework catches it and marks the test as a failed test. We are able to write an AppMap for the failed test after the failed test finishes, as usual. However, our code which creates the http response return event does not get chance to run because of this exception. This produces a bad AppMap with an http request call event missing a matching http response return event.